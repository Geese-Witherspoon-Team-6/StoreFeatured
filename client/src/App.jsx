import React from 'react';
import reactDOM from'react-dom';
import Seller from './Seller.jsx';
import ProductList from './ProductList.jsx';
import axios from 'axios';
import styled from 'styled-components';

const SellerStyle = styled.section`
  padding: 4px;
  margin: 4px;
  border: 1px solid gray;
  width: 35%;
  color: palevioletred;
`;

const Container = styled.section`
  padding: 4px;
  margin: 4px;
  border: 1px solid gray;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
`;

const testValue = null;

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      seller:null,
      products: null,
    }
  }


  componentDidMount() {
      axios.get('/api/seller', {
        params: {
          sellerName: 'Mincing Mockingbird'
        }
      })
        .then((response) => {
          let sellerInfo = response.data[0];
          let productList = response.data[0].products;
          console.log(productList)
          this.setState({
            seller: sellerInfo,
            products: productList
          })
        })
        .catch((error) => {
          console.log(error)
        })
  }

  render() {
    return (
      <div>
        <Container>
            <SellerStyle>
              <div>
                {
                  this.state.seller === null ? (
                    <>Loading...</>
                  ) : (
                    <Seller sellerInfo={this.state.seller}/>
                  )
                }
              </div>
            </SellerStyle>
          <div>
          {
            this.state.products === null ? (
              <>Loading...</>
            ) : (
              <ProductList products={this.state.products}/>
            )
          }
          </div>
        </Container>
      </div>
    )
  }
}

export default App;