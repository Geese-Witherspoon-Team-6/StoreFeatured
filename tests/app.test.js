import React from 'react';
import renderer from 'react-test-renderer';
import App from '../client/src/App.jsx';

test('two plus two is four', () => {
  expect(2 + 2).toBe(4);
});

test('object assignment', () => {
  const data = {one: 1};
  data['two'] = 2;
  expect(data).toEqual({one: 1, two: 2});
});

test('something on the App component', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.componentDidMount();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();


})