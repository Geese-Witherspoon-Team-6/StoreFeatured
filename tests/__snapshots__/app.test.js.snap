// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`something on the App component 1`] = `
<div>
  <section
    className="sc-gtssRu fiKaKn"
  >
    <section
      className="sc-bdnylx bzYRnB"
    >
      <div>
        Loading...
      </div>
    </section>
    <div>
      Loading...
    </div>
  </section>
</div>
`;
